dvc/utils/hydra.py

```diff
@@ -51,3 +75,23 @@ def apply_overrides(path: "StrPath", overrides: List[str]) -> None:
 
         merge_dicts(original_data, new_data)
         remove_missing_keys(original_data, new_data)
+
+
+def to_hydra_overrides(path_overrides):
+    parser = OverridesParser.create()
+    return parser.parse_overrides(overrides=path_overrides)
+
+
+def get_hydra_sweeps(path_overrides, hydra_output_file):
+    hydra_overrides = to_hydra_overrides(path_overrides.pop(hydra_output_file))
+    for override in hydra_overrides:
+        if override.value_type == ValueType.GLOB_CHOICE_SWEEP:
+            raise InvalidArgumentError(
+                f"Glob override '{override.input_line}' is not supported."
+            )
```

Comment:
Supporting `glob` is a little mess compared to the other syntaxes because it requires providing a config loader and couldn't find a clean way of doing this for now.
Considered the rest of the syntaxes good enough as a first step and wait for user requests on this to prioritize.

Reply:
For a large grid search or use case like https://github.com/iterative/dvc/issues/7891#issuecomment-1239800734, writing all parameter combinations to the Hydra config dir and then using glob to sweep over them all like in https://hydra.cc/docs/patterns/configuring_experiments/#sweeping-over-experiments could be pretty useful.